docs/reference/query-languages/esql/esql-lookup-join.md

```diff
@@ -87,7 +88,7 @@ FROM Left
 
 To use `LOOKUP JOIN`, you must have:
 
-* Data types of join key and join field in the lookup index need to generally be the same - up to widening of data types, where e.g. `short,byte` are considered equal to `integer`. Also, text fields can be used on the left hand side if and only if there is an exact subfield whose name is suffixed with `.keyword`.
+* Data types of join key and join field in the lookup index need to generally be the same. As long as the field data type is structurally capable of handling the data that you are passing through. For example `short,byte` is considered equal to `integer`. Also, text fields can be used on the left hand side if and only if there is an exact subfield whose name is suffixed with `.keyword`.
 
 ## Limitations
```

Comment:
Another limitation: only one join key field can be used for now, and we can only join on equality instead of more general conditions.

Reply:
Sorry, this is already mentioned.